src/event/ngx_event_openssl.c

```diff
@@ -1576,6 +1577,35 @@ ngx_ssl_send_chain(ngx_connection_t *c, ngx_chain_t *in, off_t limit)
 
     for ( ;; ) {
 
+        /* Dynamic record resizing : 
+         * We want the initial records to fit into one tcp segment
+         * so we don't get TCP HoL blocking due to TCP Slow Start.
+         * A connect always starts with sall records which fit into on tcp segment,
+         * but after a given amount of records sent, we make the records larger to reduce header overhead.
+         * After a connection has idled for a given timeout,begin the process from the start. The actual paramenters
```

Comment:
Small records incur overhead, large records incur latency, and there is no one value for the "optimal" record size. Instead, for web applications, which are consumed by the browser, the best strategy is to dynamically adjust the record size based on the state of the TCP connection:

- When the connection is new and TCP congestion window is low, or when the connection has been idle for some time ( Slow-Start Restart), each TCP packet should carry exactly one TLS record, and the TLS record should occupy the full maximum segment size (MSS) allocated by TCP.
- When the connection congestion window is large and if we are transferring a large stream (e.g., streaming video), the size of the TLS record can be increased to span multiple TCP packets (up to 16KB) to reduce framing and CPU overhead on the client and server.